sdk/storage/Azure.Storage.DataMovement.Blobs/src/AppendBlobStorageResource.cs

```diff
@@ -94,8 +90,14 @@ public AppendBlobStorageResource(AppendBlobClient blobClient, AppendBlobStorageR
             CancellationToken cancellationToken = default)
         {
             Response<BlobDownloadStreamingResult> response = await BlobClient.DownloadStreamingAsync(
-                _options.ToBlobDownloadOptions(new HttpRange(position, length), _etagDownloadLock),
+                _options.ToBlobDownloadOptions(new HttpRange(position, length), ResourceProperties?.ETag),
                 cancellationToken).ConfigureAwait(false);
+
+            // Set the resource properties if we currently do not have any stored on the resource.
+            if (ResourceProperties != null)
```

Comment:
`ResourceProperties == null` right? Also, maybe `default`? Admittedly I don't really understand when to use `null` versus `default` but seems the pattern is `default` here.

Reply:
You're right. Going to change back to `default` to be consistent. (I think I was just feeling like putting null there for some reason).

(going to also apply this to the rest of the storage resource types).